runtime/src/accounts.rs

```diff
@@ -278,6 +278,14 @@ impl Accounts {
                             })
                             .unwrap_or_default();
 
+                        if account.executable()
+                            && demote_program_write_locks
+                            && message.is_writable(i, demote_program_write_locks)
+                        {
+                            error_counters.invalid_account_index += 1;
+                            return Err(TransactionError::InvalidAccountIndex);
+                        }
```

Comment:
Does this mean we can't write lock upgradeable programs during upgrades? I think we need to add a check to see if the upgradeable loader is present

Reply:
Oh duh, I even said that myself in the slack thread. Thanks for catching me!

Reply:
Fixed in updated https://github.com/solana-labs/solana/pull/19629/commits/08eed57dd468c7f9d8ea6058e09fe58729652234